engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/callactivity/CallActivityActivatedHandler.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.handlers.callactivity;
+
+import static io.zeebe.util.buffer.BufferUtil.bufferAsString;
+
+import io.zeebe.engine.processor.KeyGenerator;
+import io.zeebe.engine.processor.workflow.BpmnStepContext;
+import io.zeebe.engine.processor.workflow.deployment.model.element.ExecutableCallActivity;
+import io.zeebe.engine.processor.workflow.handlers.element.ElementActivatedHandler;
+import io.zeebe.engine.state.deployment.DeployedWorkflow;
+import io.zeebe.protocol.impl.record.value.workflowinstance.WorkflowInstanceRecord;
+import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import io.zeebe.protocol.record.value.ErrorType;
+
+public class CallActivityActivatedHandler extends ElementActivatedHandler<ExecutableCallActivity> {
+
+  private final KeyGenerator keyGenerator;
+
+  private final WorkflowInstanceRecord childInstanceRecord = new WorkflowInstanceRecord();
+
+  public CallActivityActivatedHandler(KeyGenerator keyGenerator) {
+    super(null);
+    this.keyGenerator = keyGenerator;
+  }
+
+  @Override
+  protected boolean handleState(BpmnStepContext<ExecutableCallActivity> context) {
+    super.handleState(context);
+
+    final var processId = context.getElement().getProcessId();
+    final var workflow = context.getStateDb().getLatestWorkflowVersionByProcessId(processId);
+
+    if (workflow == null) {
```

Comment:
This check is duplicated in the `CallActivityActivatingHandler`, wouldn't it be enough to do it once when activating?

Reply:
True. I had the other case in mind that the process id is defined by an expression (which could change in between). 

Instead of raising an incident, I could throw an exception. But I would stay with the check itself to be safe (you never know).